### PR TITLE
NHIF search by category and location

### PR DESCRIPTION
**Pull request by andela-ookoro**
#### What does this PR do?
Expand the search panel for NHIF facilities to accommodate different NHIF categories.

#### Description of Task to be completed?
Add list box for NHIF category.
Edit the gradNHIFDetaiils to return dynamic result per NHIF details.
Edit the result modal to show the number of records found.
Add a style class for the span that shows the number of records found.

#### Screenshots 
Screenshot of the panel when the page loads.
panel_before_user_select

Screenshot of the panel when a user select options.
user_select_options

Screenshot of a validation note.
note_to_select_option

Screenshot of the panel when no record was found.
search_without_research

Screenshot of the panel when records are found.
search_with_results

### CODE REVIEW

File: js/healthtools.js [hunk 1]
```diff
@@ -71,6 +71,9 @@ $(document).ready(function() {
       url: url,
       success: function(response) {
         var response_html = ''
+
+
+
```
Comment (DavidLemayian) [inline js/healthtools.js]: ❓ ☝️

File: _includes/modals.html [hunk 2]
```diff
@@ -6,8 +6,9 @@
                 <div class="modal-header">
                     <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
                     <h3 id="dname"></h3>
+                    <span id="number_found"></span> 
                 </div>
-                <div class="modal-body" style="height:600px;overflow-y:scroll;">
+                <div class="modal-body" style="max-height:500px;overflow-y:scroll;">
```
Comment (DavidLemayian) [inline _includes/modals.html:11]: 🤔

File: js/healthtools.js [hunk 3]
```diff
@@ -94,7 +96,6 @@ $(document).ready(function() {
         } else {
           // Clinical Officers
           for (var k = 0; k < result_no; k++) {
-            console.log(result.hits[k])
```
Comment (DavidLemayian) [inline js/healthtools.js:97]: 💯

Comment (DavidLemayian) [inline js/healthtools.js:97]: 💯

File: js/healthtools.js [hunk 4]
```diff
@@ -127,33 +128,106 @@ $(document).ready(function() {
...
+      response_html += '</p>';
+      $('#mybox').html(response_html);
+      $('#number_found').html('');
+      $('#hospital_location').val('');
+      $('#loading').hide();
+    } else {
+      $('#dname').html('<h4>' + search_type_text + ' in ' + hospital_location + '</h4>');
+      $('#mybox').html('');
+      $('#loading').show();
+      var api_url = `https://api.healthtools.codeforafrica.org/search/${search_type}?q=${hospital_location}`;
```
Comment (DavidLemayian) [inline js/healthtools.js]: Check quotes being used  (`) vs (').

File: js/healthtools.js [hunk 5]
```diff
@@ -127,33 +128,106 @@ $(document).ready(function() {
...
+            response_html += '</p><p style="text-align: center;">';
+            response_html += '<small><em><a href="mailto:starhealth@codeforkenya.org" target="_blank">E-mail us</a></em></small>';
+            response_html += '</p>';
+          }
+
+          // Google Analytics Events
+          ga('send', 'event', 'InsuranceHospital', 'search', hospital_location, result.hits.found);
+          ga('theStar.send', 'event', 'InsuranceHospital', 'search', hospital_location, result.hits.found);
+          ga('theStarHealth.send', 'event', 'InsuranceHospital', 'search', hospital_location, result.hits.found);
+          ga('CfAFRICA.send', 'event', 'InsuranceHospital', 'search', hospital_location, result.hits.found);
```
Comment (DavidLemayian) [inline js/healthtools.js]: 'InsuranceHospital' to 'NHIF'

File: js/healthtools.js [hunk 6]
```diff
@@ -127,33 +128,106 @@ $(document).ready(function() {
...
+
+          $('#mybox').html(response_html);
+          $('#hospital_location').val('');
+          $('#loading').hide();
+        },
+        error: function(error) {
+          $('#dname').html('<h4>Note</h4>' );
+          response_html = '<p style="text-align: center;">';
+          response_html +=  'An error occurred, please try again or end us a mail';
```
Comment (DavidLemayian) [inline js/healthtools.js]: *send

**Review by DavidLemayian — CHANGES_REQUESTED** [review]
Left some comments and changes.

File: _includes/modals.html (hunk 2 above)
Comment (andela-ookoro) [reply _includes/modals.html:11]: The fixed height produced a long space with small query result, using a max-height would make the modal height responsive and would show a scroll bar only when the max-height is exceeded.

**Pull request merged**